arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -143,71 +143,59 @@ struct TimeTracker {
   TimeTracker(TimeTracker const&) = delete;
   TimeTracker& operator=(TimeTracker const&) = delete;
 
-  explicit TimeTracker(arangodb::TransactionStatistics const& statistics, 
-                       std::optional<std::reference_wrapper<arangodb::metrics::Histogram<arangodb::metrics::LogScale<float>>>>& histogram) noexcept
-      : statistics(statistics),
-        histogram(histogram) {
-    if (statistics._exportReadWriteMetrics) {
+  using Metrics = arangodb::TransactionStatistics::ReadWriteMetrics;
+  using Count = void (*)(Metrics& metrics, float time);
+
+  explicit TimeTracker(std::optional<Metrics>& metrics, Count count) noexcept
+      : metrics{metrics}, count{count} {
+    if (metrics) {
       // time measurement is not free. only do it if metrics are enabled
-      start = getTime();
+      start = std::chrono::steady_clock::now();
     }
   }
 
   ~TimeTracker() {
-    if (statistics._exportReadWriteMetrics) {
-      // metrics collection is not free. only do it if metrics are enabled
-      // unit is seconds here
-      TRI_ASSERT(histogram.has_value());
-      histogram->get().count(std::chrono::duration<float>(getTime() - start).count());
+    if (metrics) {
+      // metrics collection is not free. only do it if metrics are enabled unit is seconds here
```

Comment:
I know the code comment was there before, but I had some trouble understanding it when reading it right now. So I suggest to slightly change the wording:
```suggestion
      // metrics gathering is not free. only do it if metrics are enabled. Unit is seconds here
```

Reply:
Done